eng/common/pipelines/templates/steps/cosmos-emulator.yml

```diff
@@ -3,93 +3,34 @@ parameters:
-      }
-      $process = Start-Process $Emulator -ArgumentList "/getstatus" -PassThru -Wait
-      switch ($process.ExitCode) {
-        1 {
-          Write-Host "The emulator is already starting"
-          return
-        }
-        2 {
-          Write-Host "The emulator is already running"
-          return
-        }
-        3 {
-          Write-Host "The emulator is stopped"
-        }
-        default {
-          Write-Host "Unrecognized exit code $process.ExitCode"
-          return
-        }
-      }
-      $argumentList = ""
-      if (-not [string]::IsNullOrEmpty("${{ parameters.StartParameters }}")) {
-          $argumentList += , "${{ parameters.StartParameters }}"
-      } else {
-        # Use the default params if none provided
-        $argumentList = "/noexplorer /noui /enablepreview /disableratelimiting /enableaadauthentication"
-      }
-      Write-Host "Starting emulator process: $Emulator $argumentList"
-      $process=Start-Process $Emulator -ArgumentList $argumentList -ErrorAction Stop -PassThru
-      Write-Host "Emulator process started: $($process.Name), $($process.FileVersion)"
-      $Timeout = 600
-      $result="NotYetStarted"
-      $complete = if ($Timeout -gt 0) {
-        $start = [DateTimeOffset]::Now
-        $stop = $start.AddSeconds($Timeout)
-        {
-          $result -eq "Running" -or [DateTimeOffset]::Now -ge $stop
-        }
-      }
-      else {
-        {
-          $result -eq "Running"
-        }
-      }
-      do {
-        $process = Start-Process $Emulator -ArgumentList "/getstatus" -PassThru -Wait
-        switch ($process.ExitCode) {
-          1 {
-            Write-Host "The emulator is starting"
-          }
-          2 {
-            Write-Host "The emulator is running"
-            $result="Running"
-            return
-          }
-          3 {
-            Write-Host "The emulator is stopped"
-          }
-          default {
-            Write-Host "Unrecognized exit code $process.ExitCode"
-          }
-        }
-        Start-Sleep -Seconds 5
-      }
-      until ($complete.Invoke())
-      Write-Error "The emulator failed to reach Running status within ${Timeout} seconds"
-    displayName: Start Cosmos DB Emulator
\ No newline at end of file
+  - task: Powershell@2
+    inputs:
+      filePath: $(Build.SourcesDirectory)/eng/common/scripts/Cosmos-Emulator.ps1
+      arguments: >
+        -EmulatorMsiUrl "${{ parameters.EmulatorMsiUrl }}"
+        -StartParameters "${{ parameters.StartParameters }}"
+        -Stage "Install"
+      pwsh: true
+    displayName: Install Public Cosmos DB Emulator
+
+  - task: Powershell@2
+    inputs:
+      filePath: $(Build.SourcesDirectory)/eng/common/scripts/Cosmos-Emulator.ps1
+      arguments: >
+        -EmulatorMsiUrl "${{ parameters.EmulatorMsiUrl }}"
+        -StartParameters "${{ parameters.StartParameters }}"
+        -Stage "Launch"
+      pwsh: true
+    displayName: Launch Public Cosmos DB Emulator
+    continueOnError: true
```

Comment:
@chidozieononiwu Do you know why you left continueOnError set to true here? I'm helping @kushagraThapar look at an issue right now where the emulator isn't starting https://dev.azure.com/azure-sdk/internal/_build/results?buildId=1670972&view=logs&j=afed5e70-451a-5e51-5c23-994c4cc5c5d5&t=a14e11a5-8dc1-5518-0067-dad0c66abb33. Also with this set to continueOnError your next step will never run because we never hit a failure here.

Reply:
@milismsft  - seems like emulator is getting stuck and is not launching, and we are going to make a change of failing the CI if emulator doesn't launches. Can you please help why it can get stuck and what should be our next steps if it doesn't launch or if it fails on any of the steps?

Reply:
Does it reproduce frequently? In general emulator getting stuck is probably something to do with the environment.